plugins/DigitalLibrary/src/DigitalFactoryOutputDevice.py

```diff
@@ -105,8 +105,10 @@ def _onUserAccessStateChanged(self, logged_in: bool) -> None:
         self.enabled = logged_in and self._controller.userAccountHasLibraryAccess()
         self.enabledChanged.emit()
 
-    def _onWriteStarted(self) -> None:
+    def _onWriteStarted(self, new_name: Optional[str] = None) -> None:
         self._writing = True
+        if new_name:
+            self.setLastOutputName(new_name)  # On saving, the user can change the name, this should propagate.
```

Comment:
This change causes DigitalLibrary to require a higher API version number, meaning it's no longer backwards compatible with Cura 4.9.